light-blockchain/src/push.rs

```diff
@@ -356,4 +384,48 @@ impl LightBlockchain {
 
         Ok(PushResult::Extended)
     }
+
+    fn detect_forks(&self, block: &MicroBlock, prev_vrf_seed: &VrfSeed) {
+        assert!(!block.is_skip_block());
+
+        // Check if there are two blocks in the same slot and with the same height. Since we already
+        // verified the validator for the current slot, this is enough to check for fork proofs.
+        // Note: We don't verify the justifications for the other blocks here, since they had to
+        // already be verified in order to be added to the blockchain.
```

Comment:
```suggestion
        // Note: We don't verify the justifications for the other blocks here, since they already
        // had to be verified in order to be added to the blockchain.
```